cns/types/types.go

```diff
@@ -624,6 +624,7 @@ type CnsVolumeACLConfigureSpec struct {
 type CnsNFSAccessControlSpec struct {
 	types.DynamicData
 	Permission []vsanfstypes.VsanFileShareNetPermission `xml:"netPermission,omitempty,typeattr"`
+	Delete     bool                                     `xml:"delete,omitempty"`
```

Comment:
Can we add a reference test to revoke/remove permissions for a given volume?

Reply:
Makes sense! Let me add that

Reply:
```
$ go test -v
=== RUN   TestClient
    client_test.go:682: Creating CNS file volume using the spec: {DynamicData:{} Name:pvc-file-share-volume VolumeType:FILE Datastores:[Datastore:datastore-42] Metadata:{DynamicData:{} ContainerCluster:{DynamicData:{} ClusterType:KUBERNETES ClusterId:demo-cluster-id VSphereUser:Administrator@vsphere.local ClusterFlavor:VANILLA ClusterDistribution:OpenShift} EntityMetadata:[] ContainerClusterArray:[{DynamicData:{} ClusterType:KUBERNETES ClusterId:demo-cluster-id VSphereUser:Administrator@vsphere.local ClusterFlavor:VANILLA ClusterDistribution:OpenShift}]} BackingObjectDetails:0xc0003689c0 Profile:[] CreateSpec:0xc000364c80}
    client_test.go:707: Fileshare volume created sucessfully. filevolumeId: file:d71f42e7-ba28-4fa7-9d95-1fef4c43978d
    client_test.go:741: Invoking ConfigureVolumeACLs using the spec: types.CnsVolumeACLConfigureSpec{
            VolumeId: types.CnsVolumeId{
                Id: "file:d71f42e7-ba28-4fa7-9d95-1fef4c43978d",
            },
            AccessControlSpecList: []types.CnsNFSAccessControlSpec{
                {
                    Permission: []types.VsanFileShareNetPermission{
                        {Ips:"192.168.124.2", Permissions:"READ_ONLY", AllowRoot:false},
                    },
                    Delete: false,
                },
            },
        }
    client_test.go:782: Invoking ConfigureVolumeACLs using the spec: types.CnsVolumeACLConfigureSpec{
            VolumeId: types.CnsVolumeId{
                Id: "file:d71f42e7-ba28-4fa7-9d95-1fef4c43978d",
            },
            AccessControlSpecList: []types.CnsNFSAccessControlSpec{
                {
                    Permission: []types.VsanFileShareNetPermission{
                        {Ips:"192.168.124.2", Permissions:"READ_ONLY", AllowRoot:false},
                    },
                    Delete: true,
                },
            },
        }
--- PASS: TestClient (46.57s)
PASS
ok      github.com/vmware/govmomi/cns   46.589s
chethanv-a01:cns chethanv$ go test -v
=== RUN   TestClient
    client_test.go:682: Creating CNS file volume using the spec: {DynamicData:{} Name:pvc-file-share-volume VolumeType:FILE Datastores:[Datastore:datastore-42] Metadata:{DynamicData:{} ContainerCluster:{DynamicData:{} ClusterType:KUBERNETES ClusterId:demo-cluster-id VSphereUser:Administrator@vsphere.local ClusterFlavor:VANILLA ClusterDistribution:OpenShift} EntityMetadata:[] ContainerClusterArray:[{DynamicData:{} ClusterType:KUBERNETES ClusterId:demo-cluster-id VSphereUser:Administrator@vsphere.local ClusterFlavor:VANILLA ClusterDistribution:OpenShift}]} BackingObjectDetails:0xc000453640 Profile:[] CreateSpec:0xc000454c20}
    client_test.go:707: Fileshare volume created sucessfully. filevolumeId: file:166c3e5a-6628-4a54-b0ce-19fc142fb76e
    client_test.go:741: Invoking ConfigureVolumeACLs using the spec: types.CnsVolumeACLConfigureSpec{
            VolumeId: types.CnsVolumeId{
                Id: "file:166c3e5a-6628-4a54-b0ce-19fc142fb76e",
            },
            AccessControlSpecList: []types.CnsNFSAccessControlSpec{
                {
                    Permission: []types.VsanFileShareNetPermission{
                        {Ips:"192.168.124.2", Permissions:"READ_ONLY", AllowRoot:false},
                    },
                    Delete: false,
                },
            },
        }
--- PASS: TestClient (34.13s)
PASS
ok      github.com/vmware/govmomi/cns   34.152s
```